### PR TITLE
Implement VueX using vuex-class-modules

### LINKED ISSUE #1196: Split Spark store into independent modules

### PR DESCRIPTION
**Pull request by steersbob**
Each Spark service now has its own store module.
Removed all intermediate anonymous functions from getters.
Uses arrays as base data types in stores. Lookup is done using array.find().

Removes Session View
Removes block groups support.

Resolves #1196

### CODE REVIEW

File: src/pages/DashboardPage.vue [hunk 1]
```diff
@@ -129,20 +119,24 @@ export default class DashboardPage extends Vue {
   }
 
   changeDashboardId(): void {
```
Comment (elcojacobs) [inline src/pages/DashboardPage.vue]: This starts a dialog that eventually calls another function named changeDashboardId. The same function name for these 2 different but related functions is a bit confusing.

Comment (elcojacobs) [reply src/pages/DashboardPage.vue]: perhaps a function name that suggests this opens a dialog?

File: src/plugins/builder/store/index.ts [hunk 2]
```diff
@@ -1,127 +1,92 @@
...
 
-import { objReducer } from '@/helpers/functional';
+import { filterById } from '@/helpers/functional';
 import store from '@/store';
 
 import { BuilderLayout, PartSpec } from '../types';
 import api from './api';
 
-const fallbackSpec: PartSpec = {
+const fallbackSpec = (): PartSpec => ({
```
Comment (elcojacobs) [inline src/plugins/builder/store/index.ts:9]: :+1: default data as functions is indeed a better

**Review by elcojacobs — COMMENTED** [review]
syntax seems better and memory use is reduced.

File: src/pages/DashboardPage.vue (hunk 1 above)
Comment (steersbob) [reply src/pages/DashboardPage.vue]: Typically everything that opens a dialog is prefixed with `start`. 
Changed functions to `editDashboardId -> startChangeDashboardId -> execDashboardIdChange`.

Comment (elcojacobs) [reply src/pages/DashboardPage.vue]: Better :)

File: src/components/sidebar/ServiceIndex.vue [hunk 3]
```diff
@@ -32,11 +32,12 @@ export default class ServiceIndex extends Vue {
   }
 
   get services(): Service[] {
+    // Avoid modifying the store object
```
Comment (elcojacobs) [inline src/components/sidebar/ServiceIndex.vue:35]: Thanks for clarifying this, I needed this comment :)

**Review by elcojacobs — APPROVED** [review]

**Pull request merged**